odoo/addons/base/models/ir_fields.py

```diff
@@ -408,7 +408,7 @@ def db_id_for(self, model, field, subfield, value):
                 if len(ids) > 1:
                     warnings.append(ImportWarning(
                         _(u"Found multiple matches for value '%s' in field '%%(field)s' (%d matches)")
-                        % (value, len(ids))))
+                        % (value.replace('%', '%%') if isinstance(value, str) else value, len(ids))))
```

Comment:
```suggestion
                        % (str(value).replace('%', '%%'), len(ids))))
```
Since we're ultimately stringifying the value, it doesn't seem necessary to add a complicated dispatching, just stringify before escaping.